dataset/test/buckets_test.cpp

```diff
@@ -24,6 +24,44 @@ class DataArrayBucketTest : public ::testing::Test {
   Variable var{std::make_unique<Model>(indices, Dim::X, buffer)};
 };
 
+TEST_F(DataArrayBucketTest, concatenate_dim_1d) {
+  Variable expected_indices =
+      makeVariable<std::pair<scipp::index, scipp::index>>(
+          Values{std::pair{0, 4}});
+  Variable expected{std::make_unique<Model>(expected_indices, Dim::X, buffer)};
+  EXPECT_EQ(buckets::concatenate(var, Dim::Y), expected);
+}
+
+TEST(DataArrayBucket2dTest, concatenate_dim_2d) {
```

Comment:
Unrelated to this work but how many dimensions are supported for buckets? Can you have 3d or 4d buckets?

Reply:
Yes, you can. The limit is the (arbitrarily chosen) limit we set somewhere in `class Dimensions` affecting the max dims of variables. I think it is 6, but this could be increased if required.